presto-spi/src/main/java/com/facebook/presto/spi/ModularHashingNodeProvider.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spi;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.facebook.presto.spi.StandardErrorCode.NO_NODES_AVAILABLE;
+import static java.util.Collections.unmodifiableList;
+
+public class ModularHashingNodeProvider<T>
+        implements NodeProvider<T>
+{
+    private final List<T> sortedCandidates;
+
+    public ModularHashingNodeProvider(List<T> sortedCandidates)
```

Comment:
This might be answered in future commits, but so far, I don't see why we would need this constructor. Maybe we should just pass the list of candidates to `get()`?